clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -422,7 +422,7 @@ public void onHeartbeatSuccess(ConsumerGroupHeartbeatResponseData response) {
         if (state == MemberState.LEAVING) {
             log.debug("Ignoring heartbeat response received from broker. Member {} with epoch {} is " +
                     "already leaving the group.", memberId, memberEpoch);
-            return;
+//            return;
```

Comment:
Is there a need to process the leave group response now? We were intentionally avoiding processing the leave response just to avoid undesired effects on the client due to unexpected responses. Processing the HB response leads to updating memberId, epoch, reconciling assignments, state transitions, etc, depending on the content of the response. On the leave, we don't expect any, and we actually do the transitions that are needed the moment the leave request is sent out, so just short-circuiting here.